documentation/api/io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener.adoc

```diff
@@ -0,0 +1,217 @@
+          - broker: 2
+            host: broker-2.myingress.com
+    #...
+----
+
+[id='property-generic-listeners-{context}']
+.`listeners`
+You configure Kafka broker listeners using the `listeners` property in the `Kafka` resource.
+Each listener is defined as an array.
+
+.Example listener configuration
+[source,yaml,subs="+attributes"]
+----
+listeners:
+  - name: plain
+    port: 9092
+    type: internal
+    tls: false
+----
+
+The name and port must be unique within the Kafka cluster.
+The name can be up to 25 characters long, comprising lower-case letters and numbers.
+
+By specifying a unique name and port for each listener,
+you can configure multiple listeners.
+
+.`type`
+
+The type is set as `internal`,
+or for external listeners, as `route`, `loadbalancer`, `nodeport` or `ingress`.
+
+internal:: You can configure internal listeners with or without encryption using the `tls` property.
+route:: Configures an external listener to expose Kafka using OpenShift `Routes` and the HAProxy router.
++
+A dedicated `Route` is created for every Kafka broker pod.
+An additional `Route` is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Routes` to connect to Kafka on port 443.
+ingress:: Configures an external listener to expose Kafka using Kubernetes `Ingress` and the {NginxIngressController}.
++
+A dedicated `Ingress` resource is created for every Kafka broker pod.
+An additional `Ingress` resource is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Ingress` resources to connect to Kafka on port 443.
++
+You must specify the hostnames used by the bootstrap and per-broker services
+using xref:type-GenericKafkaListenerConfigurationBootstrap-reference[`GenericKafkaListenerConfigurationBootstrap`]
+and xref:type-GenericKafkaListenerConfigurationBroker-reference[`GenericKafkaListenerConfigurationBroker`] properties.
++
+NOTE: External listeners using `Ingress` are currently only tested with the {NginxIngressController}.
+
+loadbalancer:: Configures an external listener to expose Kafka `Loadbalancer` type `Services`.
++
+A new loadbalancer service is created for every Kafka broker pod.
+An additional loadbalancer is created to serve as a Kafka _bootstrap_ address.
+Loadbalancers listen to connections on port 9094.
+nodeport:: Configures an external listener to expose Kafka using `NodePort` type `Services`.
++
+Kafka clients connect directly to the nodes of Kubernetes.
+An additional `NodePort` type of service is created to serve as a Kafka bootstrap address.
++
+When configuring the advertised addresses for the Kafka broker pods, Strimzi uses the address of the node on which the given pod is running.
+You can use `preferredNodePortAddressType` property to configure the xref:property-listener-config-preferredNodePortAddressType-{context}[first address type checked as the node address].
++
+NOTE: TLS hostname verification is not currently supported when exposing Kafka clusters using node ports.
```

Comment:
I think it would be great to have here examples for each type. Especially with the Ingress one it would make it more obvious what is needed.

Reply:
Added. Also added source ranges + link for the loadbalancer config